### DISCUSSION TITLE
[Application] Contract verification failing with forge script

### DISCUSSION THREAD

**Discussion by hellwolf** [ethereum-optimism/developers#256] [Application Development]
### Feedback

Currently I am having trouble to verify contract on https://optimistic.etherscan.io/. I verified that it is broken by trying to deploy and verify a minimal contract using forge script. Is this issue known and tracked?

Thanks!

### Additional Information

Start verification for (1) contracts
Start verifying contract `0xD93A350c94D3E298912Ba50f3C0013bA67CEa159` deployed on optimism

Submitting verification for [src/Torex.sol:Blurb] 0xD93A350c94D3E298912Ba50f3C0013bA67CEa159.

Submitting verification for [src/Torex.sol:Blurb] 0xD93A350c94D3E298912Ba50f3C0013bA67CEa159.

Submitting verification for [src/Torex.sol:Blurb] 0xD93A350c94D3E298912Ba50f3C0013bA67CEa159.

Submitting verification for [src/Torex.sol:Blurb] 0xD93A350c94D3E298912Ba50f3C0013bA67CEa159.

Submitting verification for [src/Torex.sol:Blurb] 0xD93A350c94D3E298912Ba50f3C0013bA67CEa159.

Submitting verification for [src/Torex.sol:Blurb] 0xD93A350c94D3E298912Ba50f3C0013bA67CEa159.

Transactions saved to: /home/hellwolf/Projects/superfluid/averageX.workspace/master/packages/evm-contracts/broadcast/Deploy.s.sol/10/run-latest.json

Sensitive values saved to: /home/hellwolf/Projects/superfluid/averageX.workspace/master/packages/evm-cotracts/cache/Deploy.s.sol/10/run-latest.json

Error: 
Etherscan could not detect the deployment.

Full log: https://pastebin.com/zQFJi6ii

### logs

```
$ (source .env.d/op.sh;forge script script/Deploy.s.sol:BlurbDeployer --broadcast --verify)
[⠘] Compiling...
No files changed, compilation skipped
Script ran successfully.
 
== Logs ==
  !!! 0xD93A350c94D3E298912Ba50f3C0013bA67CEa159
 
## Setting up 1 EVM.
 
==========================
 
Chain 10
 
Estimated gas price: 0.01 gwei
 
Estimated total gas used for script: 87185
 
Estimated amount required: 0.00000087185 ETH
 
==========================
 
###
Finding wallets for all the necessary addresses...
Enter keystore password:
##
Sending transactions [0 - 0].
⠁ [00:00:00] [##########################################################################################################################] 1/1 txes (0.0s)
Transactions saved to: /home/hellwolf/Projects/superfluid/averageX.workspace/master/packages/evm-contracts/broadcast/Deploy.s.sol/10/run-latest.json
 
Sensitive values saved to: /home/hellwolf/Projects/superfluid/averageX.workspace/master/packages/evm-contracts/cache/Deploy.s.sol/10/run-latest.json
 
##
Waiting for receipts.
⠉ [00:00:06] [######################################################################################################################] 1/1 receipts (0.0s)
##### optimism
✅  [Success]Hash: 0xb18f5ab67dd12e996b5e2d75589cdbc75b700cd1766d71b131e650def37442a4
Contract Address: 0xD93A350c94D3E298912Ba50f3C0013bA67CEa159
Block: 116669257
Paid: 0.00000067072 ETH (67072 gas * 0.01 gwei)
 
 
Transactions saved to: /home/hellwolf/Projects/superfluid/averageX.workspace/master/packages/evm-contracts/broadcast/Deploy.s.sol/10/run-latest.json
 
Sensitive values saved to: /home/hellwolf/Projects/superfluid/averageX.workspace/master/packages/evm-contracts/cache/Deploy.s.sol/10/run-latest.json
 
 
 
==========================
 
ONCHAIN EXECUTION COMPLETE & SUCCESSFUL.
Total Paid: 0.00000067072 ETH (67072 gas * avg 0.01 gwei)
##
Start verification for (1) contracts
Start verifying contract `0xD93A350c94D3E298912Ba50f3C0013bA67CEa159` deployed on optimism
 
Submitting verification for [src/Torex.sol:Blurb] 0xD93A350c94D3E298912Ba50f3C0013bA67CEa159.
 
Submitting verification for [src/Torex.sol:Blurb] 0xD93A350c94D3E298912Ba50f3C0013bA67CEa159.
 
Submitting verification for [src/Torex.sol:Blurb] 0xD93A350c94D3E298912Ba50f3C0013bA67CEa159.
 
Submitting verification for [src/Torex.sol:Blurb] 0xD93A350c94D3E298912Ba50f3C0013bA67CEa159.
 
Submitting verification for [src/Torex.sol:Blurb] 0xD93A350c94D3E298912Ba50f3C0013bA67CEa159.
 
Submitting verification for [src/Torex.sol:Blurb] 0xD93A350c94D3E298912Ba50f3C0013bA67CEa159.
 
Transactions saved to: /home/hellwolf/Projects/superfluid/averageX.workspace/master/packages/evm-contracts/broadcast/Deploy.s.sol/10/run-latest.json
 
Sensitive values saved to: /home/hellwolf/Projects/superfluid/averageX.workspace/master/packages/evm-contracts/cache/Deploy.s.sol/10/run-latest.json
 
Error: 
Etherscan could not detect the deployment.
```

edit: soyboy adding logs into the post

**Comment by opfocus** [comment]
It seems that there have been reports about issues with the validity of the API key. A simple verification will be conducted.

**Reply by opfocus** [reply]
Sorry, I may not be answering your question. I am just testing the api key situation.
Start verifying contract `0xb01D1383A1234f3172fA061FDB9cD413ff5f57c0` deployed on optimism

Submitting verification for [src/Counter.sol:Counter] 0xb01D1383A1234f3172fA061FDB9cD413ff5f57c0.
Submitted contract for verification:
        Response: `OK`
        GUID: `...`
        URL: https://optimistic.etherscan.io/address/0xb01d1383a1234f3172fa061fdb9cd413ff5f57c0
Contract verification status:
Response: `NOTOK`
Details: `Pending in queue`
Contract verification status:
Response: `OK`
Details: `Pass - Verified`
Contract successfully verified

**Comment by sbvegan** [comment]
Hey @hellwolf,

I found this foundry [issue](https://github.com/foundry-rs/foundry/issues/2435) that addresses your problem. Can you try the solutions suggested in this [comment](https://github.com/foundry-rs/foundry/issues/2435#issuecomment-1193878931)?

**Comment by hellwolf** [comment]
I believe this was an intermittent issue from the service, and now it has been resolved.
